### PR TITLE
Refactor2

### PR DESCRIPTION
**Pull request by neha29myob**
Clean up: Replaced for loops with lambdas; extracted methods.

### CODE REVIEW

File: src/main/java/Handlers/FileContentHandler.java [hunk 1]
```diff
@@ -18,87 +18,105 @@
...
+        return response;
+    }
+
+    public Response patchResponse(Request request, String filePath) {
+        Response response = new Response(204);
+        response.setResponseBody(request.getRequestBody());
+        try {
+            writeToFile(filePath, request.getRequestBody());
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (preethidasa) [inline src/main/java/Handlers/FileContentHandler.java:80]: If there an IOException it would still return a 204 response

**Comment by neha29myob** [comment]
What would be a better approach to replace those if- statements in FileContentHandler class?
- switch , or
- extract in different classes?

File: src/main/java/HttpRequest/RequestParser.java [hunk 2]
```diff
@@ -72,16 +65,21 @@ private static RequestMethod getRequestMethod(String requestLine) {
...
+        List<String> parameters = Arrays.asList(getQuery(path).split("&"));
+        parameters.forEach(parameter -> {
+            int keyIndex = parameter.indexOf("=");
+            createQueryParameterMap(parameterPair, parameter, keyIndex);
+        });
+
         return parameterPair;
     }
 
+    public static void createQueryParameterMap(HashMap<String, String> parameterPair, String parameter, int keyIndex) {
```
Comment (preethidasa) [inline src/main/java/HttpRequest/RequestParser.java:77]: Do these methods have tests? Would be good to have.

**Review by preethidasa — APPROVED** [review]

**Comment by preethidasa** [comment]
You can use switch statement in the FileContentHandler. You can also extract to different classes but it might be overkill for this instance as it the code in each class is very small and simple. Play around with it in another branch and see how it goes.

**Pull request merged**